components/public-api/gitpod/v1/scm.proto

```diff
@@ -0,0 +1,65 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service SCMService {
+  rpc GetToken(GetTokenRequest) returns (GetTokenResponse) {}
+  rpc GuessTokenScopes(GuessTokenScopesRequest) returns (GuessTokenScopesResponse) {}
+  rpc SearchRepositories(SearchRepositoriesRequest) returns (SearchRepositoriesResponse) {}
+  rpc ListSuggestedRepositories(ListSuggestedRepositoriesRequest) returns (ListSuggestedRepositoriesResponse) {}
+}
+
+message GetTokenRequest {
+    string host = 1;
+    string organization_id = 2;
+}
+message GetTokenResponse {
+    GitToken token = 1;
+}
+
+message GuessTokenScopesRequest {
+    string host = 1;
+    string repo_url = 2;
+    string git_command = 3;
+}
+message GuessTokenScopesResponse {
+    repeated string scopes = 1;
+    string message = 2;
+}
+
+message SearchRepositoriesRequest {
+    string organization_id = 1;
+    string search_string = 2;
+    int32 limit = 3;
+}
+message SearchRepositoriesResponse {
+    repeated SuggestedRepository repositories = 1;
+}
+
+message ListSuggestedRepositoriesRequest {
+    string organization_id = 1;
```

Comment:
We pass in the org id in order to consider and join any configurations.
Ideally we do not create a dependency from SCM to ConfigurationService (maybe the other way around).
I believe it would be better to join in the projects on the client tbh. 
But I also believe we don't want to do all these changes now, though. So treat this as loud thinking 🤔